coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/transformers/CastOperatorTransformer.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2023-2024 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino.transformers;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlBasicTypeNameSpec;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlTypeNameSpec;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+import com.linkedin.coral.common.calcite.CalciteUtil;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.common.utils.TypeDerivationUtil;
+
+import static org.apache.calcite.sql.parser.SqlParserPos.*;
+import static org.apache.calcite.sql.type.ReturnTypes.*;
+import static org.apache.calcite.sql.type.SqlTypeName.*;
+import static org.apache.calcite.sql.type.SqlTypeName.DOUBLE;
+
+
+/**
+ * The CAST operator in Trino has some differences compared to Hive, which are handled by this transformer:
+ *
+ * 1) Hive allows for casting of TIMESTAMP to DECIMAL, by converts it to unix time if the decimal format is valid
```

Comment:
`converting`

What does `decimal format is valid` mean? The input is a timestamp, and we are casting to decimal. Which "decimal" value is being validated? 

Update: reading the example below, I am guessing you are referring to the scale/precision and the fact that they can fit a unix timestamp? You may clarify this way.

Reply:
Done